gym/envs/toy_text/taxi.py

```diff
@@ -215,10 +227,139 @@ def reset(
+            return self._render_text()
+        else:
+            return self._render_gui(mode)
+
+    def _render_gui(self, mode):
+        if self.window is None:
+            pygame.init()
+            pygame.display.set_caption("Taxi")
+            if mode == "human":
+                self.window = pygame.display.set_mode(WINDOW_SIZE)
+            else:  # rgb_array
+                self.window = pygame.Surface(WINDOW_SIZE)
+        if self.taxi_imgs is None:
+            file_names = [
+                path.join(path.dirname(__file__), "img/cab_front.png"),
+                path.join(path.dirname(__file__), "img/cab_rear.png"),
+                path.join(path.dirname(__file__), "img/cab_right.png"),
+                path.join(path.dirname(__file__), "img/cab_left.png"),
+            ]
+            self.taxi_imgs = [
+                pygame.transform.scale(pygame.image.load(file_name), self.cell_size)
+                for file_name in file_names
+            ]
+        if self.passenger_img is None:
+            file_name = path.join(path.dirname(__file__), "img/passenger.png")
+            self.passenger_img = pygame.transform.scale(
+                pygame.image.load(file_name), self.cell_size
+            )
+        if self.destination_img is None:
+            file_name = path.join(path.dirname(__file__), "img/hotel.png")
+            self.destination_img = pygame.transform.scale(
+                pygame.image.load(file_name), self.cell_size
+            )
+        if self.median_horiz is None:
+            file_names = [
+                path.join(path.dirname(__file__), "img/gridworld_median_left.png"),
+                path.join(path.dirname(__file__), "img/gridworld_median_horiz.png"),
+                path.join(path.dirname(__file__), "img/gridworld_median_right.png"),
+            ]
+            self.median_horiz = [
+                pygame.transform.scale(pygame.image.load(file_name), self.cell_size)
+                for file_name in file_names
+            ]
+        if self.median_vert is None:
+            file_names = [
+                path.join(path.dirname(__file__), "img/gridworld_median_top.png"),
+                path.join(path.dirname(__file__), "img/gridworld_median_vert.png"),
+                path.join(path.dirname(__file__), "img/gridworld_median_bottom.png"),
+            ]
+            self.median_vert = [
+                pygame.transform.scale(pygame.image.load(file_name), self.cell_size)
+                for file_name in file_names
+            ]
+        if self.background_img is None:
+            file_name = path.join(path.dirname(__file__), "img/taxi_background.png")
+            self.background_img = pygame.transform.scale(
+                pygame.image.load(file_name), self.cell_size
+            )
+
+        desc = self.desc
+        background = pygame.Surface((WINDOW_SIZE[0], WINDOW_SIZE[1]))
+        background.fill((20, 24, 35))
+        self.window.blit(background, background.get_rect())
```

Comment:
Do we need to initialize another surface for the background, or can we just do `self.window.fill((20, 24, 35))` to begin with?